app/components/IdentityCard/FailedIdentityDetails.tsx

```diff
@@ -65,6 +68,38 @@ ${getOS()}`;
 
     const [copied, setCopied] = useTimeoutState(false, 2000);
 
+    useEffect(
+        () => () => {
+            setMail(undefined);
+            setConnecting(true);
+        },
+        [identityProvider.ipInfo.ipIdentity]
+    );
+
+    const loadSupportMail = useCallback(async () => {
+        let supportMail;
+        try {
+            supportMail = await getIdentityProviderMail(
+                identityProvider.ipInfo.ipIdentity
+            );
+        } catch {
+            // continue regardless of error
+        }
+        setConnecting(false);
```

Comment:
As we try to update state asynchronously, this has to be cleaned up if the component is unmounted to avoid a leak.